client-app/src/examples/contact/mobile/AppModel.ts

```diff
@@ -0,0 +1,95 @@
+import {managed, PlainObject, XH} from '@xh/hoist/core';
+import {ContactService} from '../svc/ContactService';
+import {BaseAppModel} from '../../../BaseAppModel';
+import {NavigatorModel} from '@xh/hoist/mobile/cmp/navigator';
+import directoryPanel from './DirectoryPanel';
+import detailsPanel from './details/DetailsPanel';
+import DetailsPanelModel from './details/DetailsPanelModel';
+import DirectoryPanelModel from './DirectoryPanelModel';
+import {makeObservable, observable, runInAction} from '@xh/hoist/mobx';
+import {castArray, uniq} from 'lodash';
+
+import {
+    autoRefreshAppOption,
+    sizingModeAppOption,
+    themeAppOption
+} from '@xh/hoist/mobile/cmp/appOption';
+
+export default class AppModel extends BaseAppModel {
+    static instance: AppModel;
+
+    @observable.ref tagList: string[] = [];
+    @observable.ref contacts: PlainObject[] = [];
+
+    @managed directoryPanelModel: DirectoryPanelModel;
+    @managed detailsPanelModel: DetailsPanelModel;
+
+    @managed
+    navigatorModel: NavigatorModel = new NavigatorModel({
+        track: true,
+        pages: [
+            {id: 'default', content: directoryPanel},
+            {id: 'details', content: detailsPanel}
+        ]
+    });
+
+    override getRoutes() {
+        return [
+            {
+                name: 'default',
+                path: '/contactMobile',
+                children: [
+                    {
+                        name: 'details',
+                        path: '/:id'
+                    }
+                ]
+            }
+        ];
+    }
+
+    constructor() {
+        super();
+        makeObservable(this);
+        this.directoryPanelModel = new DirectoryPanelModel(this);
+        this.detailsPanelModel = new DetailsPanelModel(this);
+    }
+
+    async updateContactAsync(id: string, data: PlainObject = {}) {
+        // Mobile select only allows single selections, while desktop allows multiple
+        // This means we receive an array on the desktop picker, and a single value here.
+        // Convert single result into an array of one to keep data consistent.
+        if (data.tags) {
```

Comment:
Comment above explains this. At some point maybe mobile needs to be enhanced to support multi-select, but that seems out of scope here.